packages/expo-apple-authentication/ios/EXAppleAuthentication/EXAppleAuthenticationMappings.m

```diff
@@ -0,0 +1,64 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXAppleAuthentication/EXAppleAuthenticationMappings.h>
+#import <UMCore/UMLogManager.h>
+
+@implementation EXAppleAuthenticationMappings
+
++ (NSNumber *)exportCredentialState:(ASAuthorizationAppleIDProviderCredentialState)credentialState API_AVAILABLE(ios(13.0))
+{
+  switch (credentialState) {
+    case ASAuthorizationAppleIDProviderCredentialRevoked:
+      return @0;
+    case ASAuthorizationAppleIDProviderCredentialAuthorized:
+      return @1;
+    case ASAuthorizationAppleIDProviderCredentialNotFound:
+      return @2;
+    case ASAuthorizationAppleIDProviderCredentialTransferred:
+      return @3;
+  }
+}
+
++ (ASAuthorizationScope)importScope:(NSNumber *)scope API_AVAILABLE(ios(13.0))
+{
+  switch ([scope intValue]) {
+    case 0:
+      return ASAuthorizationScopeFullName;
+    case 1:
+      return ASAuthorizationScopeEmail;
+    default:
+      @throw [[NSException alloc] initWithName:@"ERR_INVALID_SCOPE"
```

Comment:
would like to avoid crashing the entire Expo app in development. Can we instead either reject the promise, or just ignore any bad values (i.e. just pass an empty array to `requestedScopes` if only bad values are passed in from JS)?

Reply:
+1 to rejecting the promise. You should not be able to crash the app with bad JS (even if you fork our JS libraries).